src/rust/engine/process_execution/src/remote.rs

```diff
@@ -969,8 +969,17 @@ pub fn make_execute_request(
     .output_directories
     .iter()
     .map(|p| {
+      // Note: An empty string which denotes capturing the working directory is converted to "."
+      // since Buildfarm intentionally does not handle this case. Moreover, the REAPI
+      // expressly discourages supporting empty strings for output_directories:
+      //
+      //   The special value of empty string is allowed,
+      //   although not recommended, and can be used to capture the entire working
+      //   directory tree, including inputs.
+      // https://github.com/bazelbuild/remote-apis/blob/3e816456ee28f01ab2e0abf72306c1f340c7b229/build/bazel/remote/execution/v2/remote_execution.proto#L592-L594
```

Comment:
`.` isn't mentioned in the spec either, but is the theory that it is actually more widely supported in practice? It's slightly odd, because if the intent of the spec was to discourage "capturing the whole directory" then it the subtext was probably that `.` was discouraged as well. Probably worth opening a PR to clarify.

Reply:
It might work for Buildfarm since George Censure wrote on the #buildfarm channel: `it might allow '.' as the capture` https://buildteamworld.slack.com/archives/C9C4H1SN7/p1633121527241200?thread_ts=1633107474.225400&cid=C9C4H1SN7

But that's not a true confirmation that it would work. I'm going to do some additional testing with remote-apis-testing before landing this.

Reply:
I'm going to land this PR so that I get a build on `main` that I can consume via PANTS_SHA. Wiil revert if the change does not pass remote-apis-testing.

Reply:
I was able to tested it without landing. Buildfarm still fails but in a different way. I'm going to need to investigate further in order to understand the failure. So this should not be merged yet.

Reply:
This feels like a buildfarm bug? The proto specifically says `""` is allowed and defines its semantics (and I assume discourages its use because it's weird an expensive and probably not what you want, rather than because it's not broadly implemented), and as @stuhood said, `"."` _isn't_ mentioned anywhere in the proto (and, arguably, violates the "paths should be canonical" spirit of the proto...)

Reply:
It is actually an intentional choice on Buildfarm's part. From the [discussion thread on BuildTeam Slack](https://buildteamworld.slack.com/archives/C9C4H1SN7/p1633107474225400): [`yeah, I probably didn't want that (tm) and didn't implement it`](https://buildteamworld.slack.com/archives/C9C4H1SN7/p1633119724238400?thread_ts=1633107474.225400&cid=C9C4H1SN7)

Reply:
cc @werkt